pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManagerConfig.java

```diff
@@ -103,7 +103,11 @@
   //
   private static final String MAX_PARALLEL_REFRESH_THREADS = "max.parallel.refresh.threads";
 
+  // Size of cache that holds errors.
+  private static final String ERROR_CACHE_SIZE = "error.cache.size";
+
   private final static String[] REQUIRED_KEYS = {INSTANCE_ID, INSTANCE_DATA_DIR, READ_MODE};
+  private static final long DEFAULT_ERROR_CACHE_SIZE = 100L;
```

Comment:
Can you make the default as 0 to preserve existing behavior?

Reply:
Folks in OSS will then need to set this explicitly to be able to debug. I'd rather keep a safe default that does not cause any issues. I did a quick math, with 100 stack traces, we should still be well within a few MBs in the worst case.

Reply:
ok, fair point. We will change at our end, if needed. And yes, the memory used will be relatively small, given it is a per instance cache